### PR TITLE
Material toelement

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

## Summary
`ToElement` for `Material`.

## Test it

Run the tests.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/Material_TEST.cc [hunk 1]
```diff
@@ -289,3 +289,119 @@ TEST(DOMMaterial, InvalidSdf)
...
+
+/////////////////////////////////////////////////
+TEST(DOMMaterial, ToElement)
+{
+  sdf::Material material;
+
+  material.SetAmbient(ignition::math::Color(0.1, 0.2, 0.3, 1));
+  material.SetDiffuse(ignition::math::Color(0.4, 0.5, 0.6, 0.5));
+  material.SetSpecular(ignition::math::Color(0.6, 0.4, 0.8, 0.8));
+  material.SetEmissive(ignition::math::Color(0.2, 0.7, 0.9, 0.1));
```
Comment (iche033) [inline src/Material_TEST.cc:301]: windows warnings for the above 4 lines:
```
'argument': truncation from 'double' to 'const float'
```

**Review by iche033 — COMMENTED** [review]
looks good to me, just some Windows warnings.

File: src/Material_TEST.cc (hunk 1 above)
Comment (nkoenig) [reply src/Material_TEST.cc]: Fixed in 41a620a

**Review by iche033 — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**